cluster-autoscaler/core/scale_down.go

```diff
@@ -387,13 +441,11 @@ func drainNode(node *apiv1.Node, pods []*apiv1.Pod, client kube_client.Interface
 		}
 		if allGone {
 			glog.V(1).Infof("All pods removed from %s", node.Name)
-			break
+			drainSuccessfull = true
```

Comment:
This is used in defered closure - I prefer to set it here once if we succeeded, instead of caring about cleanup in every place where we may fail.

I can add a comment here if you think it will be more readable that way.